content/en/blog/_posts/2024-02-29-cri-o-seccomp-oci-artifacts.md

```diff
@@ -0,0 +1,486 @@
+Now, let's modify the pod to apply the profile `quay.io/crio/seccomp:v1` to the
+container:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    seccomp-profile.kubernetes.cri-o.io/container: quay.io/crio/seccomp:v1
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+I have to delete and recreate the Pod, because only recreation will apply a new seccomp
+profile:
+
+```shell
+kubectl delete pod/pod
+```
+
+```console
+pod "pod" deleted
+```
+
+```shell
+kubectl apply -f pod.yaml
+```
+
+```console
+pod/pod created
+```
+
+The CRI-O logs will now indicate that the runtime pulled the artifact:
+
+```console
+WARN[…] Allowed annotations are specified for workload [seccomp-profile.kubernetes.cri-o.io]
+INFO[…] Found container specific seccomp profile annotation: seccomp-profile.kubernetes.cri-o.io/container=quay.io/crio/seccomp:v1  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Pulling OCI artifact from ref: quay.io/crio/seccomp:v1  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Trying to read profile from: /var/lib/containers/storage/overlay/c15f3eb1efd9abe0ae973e050c374988bba52c45a53f5c3cd80419a865013fa6/merged/seccomp.json  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Retrieved OCI artifact seccomp profile of len: 18853  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+```
+
+And the container is finally using the profile:
+
+```shell
+export CONTAINER_ID=$(sudo crictl ps --name container -q)
+sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp | head
+```
+
+```yaml
+{
+  "defaultAction": "SCMP_ACT_ERRNO",
+  "defaultErrnoRet": 38,
+  "architectures": [
+    "SCMP_ARCH_X86_64",
+    "SCMP_ARCH_X86",
+    "SCMP_ARCH_X32"
+  ],
+  "syscalls": [
+    {
+```
+
+The same would work for every container in the pod, if users replace the
+`/container` suffix with the reserved name `/POD`, for example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    seccomp-profile.kubernetes.cri-o.io/POD: quay.io/crio/seccomp:v1
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+{{< note >}}
+CRI-O has an option called `seccomp_use_default_when_empty` (config file) /
+`--seccomp-use-default-when-empty` (CLI), which is enabled by default and sets
+the `RuntimeDefault` profile if not specified in the pod spec. For testing the
+new annotation, either disable it or lookout for the log line: _"Found container
+specific seccomp profile annotation"_.
+{{< /note >}}
+
+### Using the annotation for a container image
+
+While specifying seccomp profiles as OCI artifacts on certain workloads is a
+cool feature, the majority of end users would like to link seccomp profiles to
+actual container images. This can be done by using the same annotation, but now
```

Comment:
```suggestion
published container images. This can be done by using a container image annotation; instead
of being applied to a Kubernetes Pod, the annotation is some metadata applied
```
?